oneargopy/Argo.py

```diff
@@ -1019,3 +1152,300 @@ def __determine_graph_step(self, ax, axis: str)-> int:
+                        parameter_columns.append(parameter + '_ADJUSTED_QC')
+                        parameter_columns.append(parameter + '_ADJUSTED_ERROR')
+                else : 
+                    print(f'WARNING: The parameter: {parameter} does not exist in the file.')
+            
+            if len(parameter_columns) > 0 : 
+                pressure = ['PRES', 'PRES_QC', 'PRES_ADJUSTED', 'PRES_ADJUSTED_QC', 'PRES_ADJUSTED_ERROR']
+                existing_parameter_columns = pressure + parameter_columns
+                return existing_parameter_columns
+            else : 
+                return None
+        
+        else : 
+            return None
+
+    
+    def __fill_float_data_dataframe(self, files: list)-> pd: 
+        """ A Function to load data into the float data dataframe.
+ 
+            :param: files : list - A list of files to read in data from.
+
+            :return: pd : Dataframe - The dataframe of float data with rows 
+                where measurements were not collected excluded.
+        """
+        if self.download_settings.verbose: print(f'Loading float data...')
+
+        # Getting the file paths for downloaded .nc files
+        directory = Path(self.download_settings.base_dir.joinpath("Profiles"))
+        file_paths = []
+        for file in files : 
+            file_paths.append(directory.joinpath(file))
+
+        # Columns that will always be in the dataframe, these columns are one dimensional
+        static_columns = ['WMOID', 'CYCLE_NUMBER', 'DIRECTION', 
+                                'DATE', 'DATE_QC', 'LATITUDE', 
+                                'LONGITUDE', 'POSITION_QC',]
+        # Columns that need to be calculated or derived 
+        special_case_static_columns = ['DATE', 'DATE_QC', 'WMOID']
+
+        # Empty Dataframe to return at end of function with all loaded data
+        float_data_dataframe = pd.DataFrame()
+
+        # Iterate through files
+        for file in file_paths :             
+            # Open File
+            nc_file = netCDF4.Dataset(file, mode='r')
+            
+            # Get dimensions of .nc file
+            number_of_profiles = nc_file.dimensions['N_PROF'].size
+            number_of_levels = nc_file.dimensions['N_LEVELS'].size
+
+            # Get float id of current file
+            float_id_array = nc_file.variables['PLATFORM_NUMBER'][0]
+            float_id_array = [elem.decode('utf-8') if isinstance(elem, bytes) else elem for elem in float_id_array]
+            float_id_array.pop() # The last value in the array is 'masked' which we don't need to form the float_id
+            float_id = int(''.join(float_id_array))
+
+            # Get the range of profiles from the index file
+            profile_count = self.prof_index['wmoid'].value_counts().get(float_id, 0)
+
+            # Compare against .nc range to index file range
+            if profile_count > number_of_profiles : 
```

Comment:
only if float_profiles_dict is not None
(if someone wants all the data, it doesn't matter if there is a mismatch)